Automattic/jetpack
# Add Comment Likes module

Resolves: https://github.com/Automattic/jetpack/issues/3224

P2 post: p7jreA-i6-p2

Related WP.com patches: ~D5668-code~, ~D5930-code~, ~D5955-code~, ~D5931-code~ (all :ship:)

Add comment likes widgets in a similar way that post likes work. In order to avoid performance issues with loading too many `iframe`s at once, we are loading and unloading them when they are scrolled in and out of view respectively. 

### Proposed changes

Adds Comment Likes module that is disabled by default. When enabled, comment like widget will be shown for each comment, allowing any logged in WordPress.com user to like it. Additionally comment like counts will be shown in comments management section of `wp-admin`.

### Testing instructions 

  1. Run `yarn build` in your Jetpack folder.
  2. Navigate to `Discussion` tab in Jetpack settings, and enable Comment likes module.
  3. Test liking and un-liking comments. 
  4. Repeat 3 while logged out of WP.com.
  5. Test performance on pages with a large number of comments.
  6. Test with different browser versions, with special emphasis on mobile.
  7. Visit `wp-admin/edit-comments.php` and verify that comment like counts are displayed correctly.
  8. Visit `wp-admin/options-general.php?page=sharing` and check whether comment likes obey these settings.
  9. Try checking/unchecking `Show Likes` option in Post editor. Comment likes should respect this setting in the same way as Post likes do.
  10. Try steps 8 and 9 when Post likes, Publicize connections, and Sharing buttons modules are disabled.
  11. Make sure that post likes still work as expected.

### Proposed changelog
Enhancement: introduces new feature to like a comment, showing how many visitors liked a comment.
